src/Controller/AgendaBroadcastController.php

```diff
@@ -0,0 +1,142 @@
+use Symfony\Component\Translation\TranslatableMessage;
+
+/**
+ * @Route("/agenda/{id}/broadcast")
+ */
+class AgendaBroadcastController extends AbstractController
+{
+    /**
+     * @Route("/index", name="agenda_broadcast", methods={"GET"})
+     */
+    public function index(Agenda $agenda, DigitalPostRepository $digitalPostRepository): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $agenda);
+
+        return $this->render('agenda/broadcast/index.html.twig', [
+            'agenda' => $agenda,
+            'digital_posts' => $digitalPostRepository->findByEntity($agenda, [], ['createdAt' => Criteria::DESC]),
+        ]);
+    }
+
+    /**
+     * @Route("/create", name="agenda_broadcast_create", methods={"GET", "POST"})
+     */
+    public function broadcastAgenda(Agenda $agenda, CprHelper $cprHelper, DigitalPostHelper $digitalPostHelper, DocumentUploader $documentUploader, EntityManagerInterface $entityManager, MailTemplateHelper $mailTemplateHelper, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $agenda);
+
+        $mailTemplates = $mailTemplateHelper->getTemplates('agenda_broadcast');
+
+        $agendaBroadcast = new AgendaBroadcast();
+
+        $form = $this->createForm(AgendaBroadcastType::class, $agendaBroadcast, [
+            'mail_template_choices' => $mailTemplates,
+        ]);
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            // Create new file from template
+            $fileName = $mailTemplateHelper->renderMailTemplate($agendaBroadcast->getTemplate(), $agenda);
+
+            // Move file
+            $documentUploader->specifyDirectory('/case_documents/');
+            $updatedFileName = $documentUploader->uploadFile($fileName);
+
+            // Create document
+            $document = new Document();
+            $document->setFilename($updatedFileName);
+            $document->setDocumentName($agendaBroadcast->getTitle());
+
+            /** @var User $user */
+            $user = $this->getUser();
+            $document->setUploadedBy($user);
+            $document->setType(new TranslatableMessage('Agenda broadcast', [], 'agenda'));
+
+            $entityManager->persist($document);
+
+            $agendaBroadcast->setDocument($document);
+
+            // Create a recipient per board member
+            $boardMembers = $agenda->getBoardmembers();
+            $digitalPostRecipients = [];
+
+            foreach ($boardMembers as $boardMember) {
+                $boardMemberAddress = $cprHelper->getAddressFromCpr($boardMember->getCpr());
+
+                $digitalPostRecipients[] = (new DigitalPost\Recipient())
+                    ->setName($boardMember->getName())
+                    ->setIdentifierType('CPR')
+                    ->setIdentifier($boardMember->getCpr())
+                    ->setAddress($boardMemberAddress)
+                ;
+            }
```

Comment:
We probably need to handle exceptions when getting the board member address.

Reply:
Handled exceptions in 830e7bd1942f6fc8d52396e7bdc1bf1cbd9b654e.